test/test_image.py

```diff
@@ -134,20 +168,26 @@ def test_write_jpeg(self):
             self.assertEqual(torch_bytes, pil_bytes)
 
     def test_decode_png(self):
+        conversion = [(None, 0), ("L", 1), ("LA", 2), ("RGB", 3), ("RGBA", 4)]
         for img_path in get_images(FAKEDATA_DIR, ".png"):
-            img_pil = torch.from_numpy(np.array(Image.open(img_path)))
-            if len(img_pil.shape) == 3:
-                img_pil = img_pil.permute(2, 0, 1)
-            else:
-                img_pil = img_pil.unsqueeze(0)
-            data = read_file(img_path)
-            img_lpng = decode_png(data)
-            self.assertTrue(img_lpng.equal(img_pil))
+            for pil_mode, channels in conversion:
+                with Image.open(img_path) as img:
+                    if not (pil_mode == "L" and img.mode == "P") and pil_mode is not None:
+                        # Don't grayscale palette images
+                        img = img.convert(pil_mode)
+                    img_pil = torch.from_numpy(np.array(img))
 
-            with self.assertRaises(RuntimeError):
-                decode_png(torch.empty((), dtype=torch.uint8))
-            with self.assertRaises(RuntimeError):
-                decode_png(torch.randint(3, 5, (300,), dtype=torch.uint8))
+                img_pil = normalize_dimensions(img_pil)
+                data = read_file(img_path)
+                img_lpng = decode_png(data, channels=channels)
+
+                tol = 0 if conversion is None else 1
+                self.assertTrue(img_lpng.allclose(img_pil, atol=tol))
```

Comment:
Adding a small tolerance on the pixel values. Small variations can be caused by implementation differences between PIL and libpng.

Reply:
Which tests do have a different tolerance? Is it for the grayscale conversion? Maybe we could make the test a bit more strict and only give a tolerance in this case?

Reply:
If you check original versions of this snippet, you will see that originally I was checking with `allclose` + tolerance only the grayscale for which I knew they have different implementations. Every other conversion was checked for equality. This works find on Linux and macOS but unfortunately not on Windows. This is why I changed this for all. As I'm not on windows I don't know which other conversion fails, but I could experiment with different values and rely on the CI to tell me which works. Thoughts?

Reply:
The failures on Windows were probably the same as what we had before, and was probably the reason why we had the `.pth` file

Reply:
Ideally it would be good to have exact checks here, as we are loading the image, so we should make sure that we are consistently loading and returning the same result across versions.
If by some reason the results are slightly different, it's technically a BC-breaking change and it would be good to catch it warn users about it during the release notes (if we need to break BC)

I don't mind resolving this conversation, but I just wanted to point out that this is probably something we would like to be a bit more strict in the future and move back to exact checks